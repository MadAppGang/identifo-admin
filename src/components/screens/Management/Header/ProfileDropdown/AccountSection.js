import React, { useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import AccountIcon from '~/components/icons/AccountIcon';

const AccountSection = ({ history, onClick }) => {
  const handleClick = () => {
    history.push('/management/settings');
    onClick();
  };

  return (
    <button
      type="button"
      className="iap-profile-dropdown__section"
      onClick={useCallback(handleClick, [history, onClick])}
    >
      <AccountIcon />
      <span>Account</span>
    </button>
  );
};

AccountSection.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  onClick: PropTypes.func,
};

AccountSection.defaultProps = {
  onClick: () => {},
};

export default withRouter(AccountSection);
