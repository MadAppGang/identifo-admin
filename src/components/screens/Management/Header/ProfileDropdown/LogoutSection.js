import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '~/modules/auth/actions';
import LogoutIcon from '~/components/icons/LogoutIcon.svg';

const LogoutSection = (props) => {
  const handleClick = () => {
    props.logout();
    props.onClick();
  };

  return (
    <button
      type="button"
      className="iap-profile-dropdown__section"
      onClick={useCallback(handleClick, [props.logout, props.onClick])}
    >
      <LogoutIcon className="iap-profile-dropdown__icon" fill="#6d6d6d" />
      <span>Logout</span>
    </button>
  );
};

LogoutSection.propTypes = {
  logout: PropTypes.func.isRequired,
  onClick: PropTypes.func,
};

LogoutSection.defaultProps = {
  onClick: () => {},
};

export { LogoutSection };

export default connect(null, { logout })(LogoutSection);
