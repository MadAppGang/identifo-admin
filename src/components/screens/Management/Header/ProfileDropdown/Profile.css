.iap-header-profile {
  position: relative;
}

.iap-header-profile__trigger {
  font-size: 15px;
  padding: 5px;
  border: none;
  background-color: transparent;
  cursor: pointer;
  display: flex;
  align-items: center;
  outline: none;
}

.iap-header-profile__trigger:hover {
  opacity: .6;
}

.iap-header-profile__trigger:focus {
  color: #1B9CFC;
}

.iap-header-profile__trigger:focus:hover {
  opacity: 1;
}

.iap-header-profile__trigger span:first-of-type {
  margin-right: 10px;
}

.iap-profile-dropdown {
  padding: 4px 0;
  background: #fff;
  box-shadow: 1px 3px 5px 1px #ddd;
  border-radius: 3px;
  position: absolute;
  bottom: 0;
  right: 0;
  transform-origin: top right;
  animation: profile-dropdown-fadeout 100ms forwards;
  z-index: 1;
  min-width: 150px;
}

.iap-profile-dropdown__section {
  width: 100%;
  border: none;
  outline: none;
  background-color: #fff;
  text-align: left;
  font: normal 15px 'Circe';
  padding: 10px 16px;
  cursor: pointer;
  color: #616161;
  display: flex;
  align-items: center;
}

.iap-profile-dropdown__section:hover,
.iap-profile-dropdown__section:focus {
  background-color: #f0f9ff;
}

.iap-profile-dropdown__section span:first-of-type {
  margin-left: 13px;
}

.iap-dropdown-icon {
  height: 11px;
  width: 11px;
  display: inline-block;
}

.iap-profile-dropdown__icon {
  height: 16px;
  width: 16px;
  display: inline-block;
  transform: translateY(-2px);
}