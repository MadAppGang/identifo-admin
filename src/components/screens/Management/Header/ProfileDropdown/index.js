import React from 'react';
import DropdownIcon from '~/components/icons/DropdownIcon';
import LogoutSection from './LogoutSection';
import Dropdown from '~/components/shared/Dropdown';
import './Profile.css';

const ProfileDropdown = () => {
  return (
    <Dropdown>
      {({ isOpen, open }) => (
        <div className="iap-header-profile">
          <button
            type="button"
            className="iap-header-profile__trigger"
            onClick={open}
          >
            <span>Admin Panel</span>
            <DropdownIcon />
          </button>
          {isOpen && (
            <div className="iap-profile-dropdown">
              <LogoutSection />
            </div>
          )}
        </div>
      )}
    </Dropdown>
  );
};

export default ProfileDropdown;
