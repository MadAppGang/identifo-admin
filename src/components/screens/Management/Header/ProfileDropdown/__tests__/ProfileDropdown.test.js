import ProfileDropdown from '..';

describe('<ProfileDropdown />', () => {
  test('renders as expected', () => {
    expect(shallow(<ProfileDropdown />).find('Dropdown').dive()).toMatchSnapshot();
  });
});
