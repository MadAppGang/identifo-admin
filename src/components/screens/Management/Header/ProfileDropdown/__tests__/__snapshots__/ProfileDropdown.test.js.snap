// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileDropdown /> renders as expected 1`] = `
<div>
  <div
    className="iap-header-profile"
  >
    <button
      className="iap-header-profile__trigger"
      onClick={[Function]}
      type="button"
    >
      <span>
        Admin Panel
      </span>
      <DropdownIcon
        className="iap-dropdown-icon"
      />
    </button>
  </div>
</div>
`;
